HOWTO-make-a-release.md

```diff
@@ -0,0 +1,348 @@
+    -   [Check the website](#check-the-website)
+    -   [Send the announcement mail](#send-the-announcement-mail)
+    -   [Unfreeze the source repository](#unfreeze-the-source-repository)
+    -   [Security fixes](#security-fixes)
+    -   [Keep in touch](#keep-in-touch)
+
+
+# Prerequisites
+
+## Check your access
+
+To perform a release, you must have appropriate access to OpenSSL's
+development host, dev.openssl.org.  To test this, try to log in with
+ssh:
+
+    ssh dev.openssl.org
+
+You must also check that you can perform tasks as the user 'openssl'
+on dev.openssl.org.  When you have successfully logged in, test your
+access to that user with sudo:
+
+    sudo -u openssl id
+
+## PGP / GnuPG key
+
+You must have a PGP / GnuPG key, and its fingerprint should be present
+in the file `doc/fingerprints.txt` in the OpenSSL source.
+
+## Repositories
+
+You must have access to the following repositories:
+
+-   `openssl-git@git.openssl.org:openssl.git`
+
+    This is the usual main source repository
+
+-   `openssl-git@git.openssl.org:openssl-web.git`
+
+    This is the website repository
+
+-   `openssl-git@git.openssl.org:tools.git`
+
+    This contains certain common tools
+
+## A method for reviewing
+
+For reviewing to take place, the release person and the reviewer need a
+way to share changes that are being applied.  Most commonly, that's done
+as PRs through Github.  However, if undisclosed security fixes are being
+applied, some other more private repository must be used.
+Worst case scenario, there have been times where the review process has
+been conducted by sending patches (made with `git format-patch`) by email
+and receiving approval as a reply.
+
+The release person and the reviewer must have a conversation to confirm
+or figure out how review shall be done.
+
+# Pre-publishing tasks
+
+All the actions in this section need to be repeated for each OpenSSL
+version released.
+
+## Freeze the source repository
+
+The day before the release, freeze the main repository.  This locks
+out everyone but the named user, who is doing the release, from doing
+any pushes.  Someone other than the person doing the release should
+run the command.  For example:
+
+    ssh openssl-git@git.openssl.org freeze openssl NAME
+
+## Prepare your repository checkouts
+
+You will need to checkout at least three working trees:
+
+-   one for the website
+
+        git clone openssl-git@git.openssl.org:openssl-web.git website
+
+-   one for extra tools
+
+        git clone openssl-git@git.openssl.org:openssl-tools.git tools
+
+-   At least one for the release to be done.  You may have to repeat
+    the release tasks that follow more than once, so it's often easier
+    to have separate copies for each release branch:
+
+        git clone openssl-git@git.openssl.org:openssl.git rel-111
+        cd rel-111
+        git branch --track OpenSSL_1_1_1-stable origin/OpenSSL_1_1_1-stable
+        git checkout OpenSSL_1_1_1-stable
```

Comment:
I never create a fresh clone, but always use a pre-existing one. I also don't have a different copy for each version as the text suggests. I'd suggest ensuring the tree is clean before you start.

Reply:
> I'd suggest ensuring the tree is clean before you start.

The new `dev/release.sh` is designed to avoid that need, FYI

Reply:
Either way, there are different ways, and it really comes down whatever the release person is comfortable with.  I usually clone multiple copies and do quite a lot in parallel.  Four xterms with mkrelease.pl running in each is quite an impressive sight, let me tell you :wink:

Reply:
Can we rely on the intelligence of the release person to do something sensible that works for them?  This is more an example of how one can do things than the mandatory set of lines.

Reply:
Yep